cert/ca.go

```diff
@@ -78,22 +79,26 @@ func PktTox509(pktJson []byte, caBytes []byte, caPkSk *ecdsa.PrivateKey, require
 	// 	return nil, err
 	// }
 
-	iss, aud, email, err := pkt.GetClaims()
-	if err != nil {
+	var payload struct {
+		Issuer   string   `json:"iss"`
+		Audience []string `json:"aud"`
+		Email    string   `json:"sub"`
+	}
+	if err := json.Unmarshal(pkt.Payload, &payload); err != nil {
 		return nil, err
 	}
 
-	if string(aud) != requiredAudience {
-		return nil, fmt.Errorf("audience 'aud' claim in PK Token did not match audience required by CA, it was %s instead", string(aud))
+	if payload.Audience[0] != requiredAudience {
+		return nil, fmt.Errorf("audience 'aud' claim in PK Token did not match audience required by CA, it was %s instead", payload.Audience)
 	}
 
 	caTemplate, err := x509.ParseCertificate(caBytes)
 	if err != nil {
 		return nil, err
 	}
 
-	subject := string(email)
-	oidcIssuer := iss
+	subject := payload.Email
```

Comment:
email != subject

Reply:
I'm not sure I understand this, this line replaces the existing line you had with the same variable name and existing functionality. I have removed the line and am just using the map object directly, maybe that solves the issue? 299f73f